app/src/main/java/org/schabi/newpipe/player/helper/PlayerHolder.java

```diff
@@ -99,30 +104,46 @@ public int getQueuePosition() {
         return player.getPlayQueue().getIndex();
     }
 
-    public void setListener(@Nullable final PlayerServiceExtendedEventListener newListener) {
-        listener = newListener;
+    public void unsetListeners() {
+        listener = null;
+        holderListener = null;
+    }
 
-        if (listener == null) {
-            return;
-        }
+    public void setListener(@NonNull final PlayerServiceEventListener newListener,
+                            @NonNull final PlayerHolderLifecycleEventListener newHolderListener) {
+        listener = newListener;
+        holderListener = newHolderListener;
 
         // Force reload data from service
         if (player != null) {
-            listener.onServiceConnected(player, playerService, false);
-            startPlayerListener();
+            holderListener.onServiceConnected(playerService, false);
+            player.setFragmentListener(internalListener);
         }
     }
 
-    // helper to handle context in common place as using the same
-    // context to bind/unbind a service is crucial
+    /** Helper to handle context in common place as using the same
+     * context to bind/unbind a service is crucial.
+     *
+     * @return the common context
+     * */
     private Context getCommonContext() {
         return App.getInstance();
     }
```

Comment:
This can be removed by using `App.getInstance()` directly.

Reply:
I think the focus here is on the fact that this is the context in common. So I'd keep it.